src/compute/filter.rs

```diff
@@ -256,12 +256,21 @@ pub fn filter_record_batch(
     record_batch: &RecordBatch,
     filter: &BooleanArray,
 ) -> Result<RecordBatch> {
-    let filter = build_filter(filter)?;
-    let filtered_arrays = record_batch
-        .columns()
-        .iter()
-        .map(|a| filter(a.as_ref()).into())
-        .collect();
+    let num_colums = record_batch.columns().len();
+
+    let filtered_arrays = match num_colums {
+        1 => {
+            vec![super::filter::filter(record_batch.columns()[0].as_ref(), filter)?.into()]
```

Comment:
```suggestion
            vec![filter(record_batch.columns()[0].as_ref(), filter)?.into()]
```

Reply:
Not possible, but I'll rename the parameter instead (also micro-bench incoming).